modules/development/src/main/java/unnamed/mmo/server/dev/Main.java

```diff
@@ -48,23 +60,38 @@ public static void main(String[] args) {
             player.setGameMode(GameMode.SURVIVAL);
             player.setPermissionLevel(4);
             player.setAllowFlying(true);
+
+            // Testing
+            event.getSpawnInstance().setBlock(5, 43, 5, Ore.fromNamespaceId("unnamed:gold_ore").asBlock());
+            player.getInventory().addItemStack(Item.fromNamespaceId("unnamed:diamond_pickaxe").asItemStack());
+
+            //todo this needs to be done elsewhere
+            player.addEffect(new Potion(PotionEffect.MINING_FATIGUE, (byte) -1, Short.MAX_VALUE, (byte) 0x0));
+
+            //todo a command for this
+            player.getInventory().addItemStack(DebugToolManager.createTool("unnamed:hello"));
         });
 
         BaseCommandRegister.registerCommands();
 
         // For now, manually register chat (with conn to mongo :/ need a config system)
-        MongoClient mongoClient = MongoClients.create(MongoClientSettings.builder()
-                .applyConnectionString(new ConnectionString("mongodb://localhost:27017"))
-                .uuidRepresentation(UuidRepresentation.STANDARD)
-                .build());
-        ChatStorage chatStorage = ChatStorage.mongo(mongoClient);
+//        MongoClient mongoClient = MongoClients.create(MongoClientSettings.builder()
+//                .applyConnectionString(new ConnectionString("mongodb://localhost:27017"))
+//                .uuidRepresentation(UuidRepresentation.STANDARD)
+//                .build());
+//        ChatStorage chatStorage = ChatStorage.mongo(mongoClient);
+        ChatStorage chatStorage = ChatStorage.noop();
         ChatManager chatManager = new ChatManager(chatStorage);
         chatManager.hook(MinecraftServer.process());
 
         //todo properly implement a config system & use facets better
         ItemManager itemManager = new ItemManager();
         itemManager.hook(MinecraftServer.process());
 
+        //todo stupid facet implementation
```

Comment:
I'd recommend some sort of Set<Facet>, and then iterate over that set to hook into the server process

So you only need to do set.add(new Facet()); and you don't need to call the hook method for each thing.

Reply:
Eventually it will be loaded using a [ServiceLoader](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwj_1Ynvmsf5AhUgArkGHUySARMQFnoECAcQAQ&url=https%3A%2F%2Fdocs.oracle.com%2Fjavase%2F8%2Fdocs%2Fapi%2Fjava%2Futil%2FServiceLoader.html&usg=AOvVaw0HhTv9ZiYFkhCrHKpIWFAX), but at the moment ChatManager cannot get its storage on its own. It need a config system. I tried a draft of this but wasn't happy so I tossed it.